client/src/ProtectedRoutes/ProtectedRoutes.jsx

```diff
@@ -11,25 +12,26 @@ import {
 
 export function ProtectedAdmin({ children }) {
   const { user } = useAuth();
-  if (user.role === ADMIN_ROLE) {
-    return <Navigate to="/dashboard" replace />;
+  if (user?.role !== ADMIN_ROLE) {
+    return <Navigate to="/" replace />;
   }
-  return children;
+  return children || <Outlet />;
 }
 export function ProtectedUser({ children }) {
   const { user } = useAuth();
   if (user) {
```

Comment:
if there is no user you should navigate to home page
```suggestion
  if (!user) {
```

Reply:
done